lib/puppet/file_serving/metadata.rb

```diff
@@ -88,6 +90,21 @@ def collect_stat(path)
   def collect(source_permissions = nil)
     real_path = full_path
 
+    if static_catalog
+      if environment && environment.static_catalogs?
+        env_path = File.join(Puppet[:environmentpath], environment.name.to_s) # environment.name is a symbol sometimes
+        puts "Checking if #{real_path} is in #{env_path}"
+        if real_path.start_with? env_path
```

Comment:
Could the inlinable logic be moved out of Metadata and into the compiler itself (since the compiler already knows the environment, and you can call `Metadata#full_path` to get the resolved path)? That way we don't have to lookup the `:environmentpath` each time we collect metadata or modify metadata serialization?